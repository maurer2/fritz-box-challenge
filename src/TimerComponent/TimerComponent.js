import React, { PureComponent } from 'react';
import styled from 'styled-components/macro';
import { PropTypes } from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';

const TimerWrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  padding: 2.5vh 2.5vw;
  text-align: center;
  font-size: 5vw;
  background: red;
`;

const FadeTransition = styled.div`
  .slide-up-appear,
  .slide-up-enter {
    transform: translateY(-100%);
  }

  .slide-up-appear.slide-up-appear-active,
  .slide-up-enter.slide-up-enter-active {
    transform: translateY(0);
    transition: transform 250ms ease-out;
  }

  .slide-up-leave {
    transform: translateY(0);
  }

  .slide-up-leave.slide-up-leave-active {
    transform: translateY(-100%);
    transition: transform 250ms ease-out;
  }
`;

class TimerComponent extends PureComponent {
  render() {
    return (
      <FadeTransition>
        <CSSTransitionGroup
          component={ React.Fragment }
          transitionAppear={ false }
          transitionName="slide-up"
          transitionAppearTimeout={ 0 }
          transitionLeaveTimeout={ 250 }
          transitionEnterTimeout={ 250 }
        >
        { this.props.isUpdating && (
          <TimerWrapper>
            Updating!
          </TimerWrapper>
        )}
        </CSSTransitionGroup>
      </FadeTransition>
    );
  }
}

TimerComponent.propTypes = {
  isUpdating: PropTypes.bool,
};

export default TimerComponent;
