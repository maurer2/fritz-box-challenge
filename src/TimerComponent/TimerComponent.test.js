import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import TimerComponent from './TimerComponent';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('TimerComponent dummy', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<TimerComponent/>);

    expect(wrapper).toMatchSnapshot();
  });
});
