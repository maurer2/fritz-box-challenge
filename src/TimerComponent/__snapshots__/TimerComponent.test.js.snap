// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerComponent dummy should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimerComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <CSSTransitionGroup
        component={Symbol(react.fragment)}
        transitionAppear={false}
        transitionAppearTimeout={0}
        transitionEnter={true}
        transitionEnterTimeout={250}
        transitionLeave={true}
        transitionLeaveTimeout={250}
        transitionName="slide"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "component": Symbol(react.fragment),
        "transitionAppear": false,
        "transitionAppearTimeout": 0,
        "transitionEnter": true,
        "transitionEnterTimeout": 250,
        "transitionLeave": true,
        "transitionLeaveTimeout": 250,
        "transitionName": "slide",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "SladeYTransition-gl3nac-1",
        "isStatic": true,
        "rules": Array [
          "position:absolute;top:0;left:0;right:0;.slide-appear,.slide-enter{transform:translateY(-100%);}.slide-appear.slide-appear-active,.slide-enter.slide-enter-active{transform:translateY(0);transition:transform 250ms ease-out;}.slide-leave{transform:translateY(0);}.slide-leave.slide-leave-active{transform:translateY(-100%);transition:transform 250ms ease-out;}",
        ],
      },
      "displayName": "SladeYTransition",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "SladeYTransition-gl3nac-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <CSSTransitionGroup
          component={Symbol(react.fragment)}
          transitionAppear={false}
          transitionAppearTimeout={0}
          transitionEnter={true}
          transitionEnterTimeout={250}
          transitionLeave={true}
          transitionLeaveTimeout={250}
          transitionName="slide"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "component": Symbol(react.fragment),
          "transitionAppear": false,
          "transitionAppearTimeout": 0,
          "transitionEnter": true,
          "transitionEnterTimeout": 250,
          "transitionLeave": true,
          "transitionLeaveTimeout": 250,
          "transitionName": "slide",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "SladeYTransition-gl3nac-1",
          "isStatic": true,
          "rules": Array [
            "position:absolute;top:0;left:0;right:0;.slide-appear,.slide-enter{transform:translateY(-100%);}.slide-appear.slide-appear-active,.slide-enter.slide-enter-active{transform:translateY(0);transition:transform 250ms ease-out;}.slide-leave{transform:translateY(0);}.slide-leave.slide-leave-active{transform:translateY(-100%);transition:transform 250ms ease-out;}",
          ],
        },
        "displayName": "SladeYTransition",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "SladeYTransition-gl3nac-1",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
